### DISCUSSION TITLE
Synchronization of CartesianLayer Addition/Removal to Transactions

### DISCUSSION THREAD

**Discussion by Tyler-Lopez** [patrykandpatrick/vico#916] [Ideas]
### Background

Hi there,

Per the conversation above in the mistakenly labeled "bug" issue, the Vico library currently lacks the functionality to dynamically add or remove `CartesianLayer` synchronously with a transaction. The issue itself was not truly a bug, as the crash reported was pointed out to be a bug of Compose as a whole.

I was made aware within that issue's discussion that other mutations within a `CartesianLayer` like adding or removing a Line from a `LineCartesianLayer`'s `LineProvider` **_are_** currently possible to do synchronously. I am not yet knowledgeable on how to do this myself.

**Comment by Gowsky** [comment]
Hello, @Tyler-Lopez. Thanks for the suggestion. We'll look into it.

> I was made aware within that issue's discussion that other mutations within a `CartesianLayer` like adding or removing a Line from a `LineCartesianLayer`'s `LineProvider` **_are_** currently possible to do synchronously. I am not yet knowledgeable on how to do this myself.

Addressed in #924.
